Dockerfile

```diff
@@ -39,65 +43,50 @@ RUN apt-get update \
     nodejs \
     npm \
  && apt-get clean \
- && rm -rf /var/lib/apt/lists/*
-
-RUN python3 -m pip install --upgrade setuptools pip build wheel
-RUN npm install --global yarn
-
-# copy everything except whats in .dockerignore, its a
-# compromise between needing to rebuild and maintaining
-# what needs to be part of the build
-COPY . /src/jupyterhub/
-WORKDIR /src/jupyterhub
+ && rm -rf /var/lib/apt/lists/* \
+ && python3 -m pip install --no-cache-dir --upgrade setuptools pip build wheel \
+ && npm install --global yarn \
+ # Build client component packages (they will be copied into ./share and
+ # packaged with the built wheel.)
+ && python3 -m build --wheel \
+ && python3 -m pip wheel --wheel-dir wheelhouse dist/*.whl
 
-# Build client component packages (they will be copied into ./share and
-# packaged with the built wheel.)
-RUN python3 -m build --wheel
-RUN python3 -m pip wheel --wheel-dir wheelhouse dist/*.whl
+FROM $BASE_IMAGE
+ENV DEBIAN_FRONTEND=noninteractive \
+    SHELL=/bin/bash \
+    LC_ALL=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    LANGUAGE=en_US.UTF-8
 
+EXPOSE 8000
 
-FROM $BASE_IMAGE
+LABEL maintainer="Jupyter Project <jupyter@googlegroups.com>"
+LABEL org.jupyter.service="jupyterhub"
 
-USER root
+WORKDIR /srv/jupyterhub
 
-ENV DEBIAN_FRONTEND=noninteractive
+COPY --from=builder /src/jupyterhub/wheelhouse /tmp/wheelhouse
```

Comment:
I think if we're optimizing image size, using [mounts as we do in other images](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/75f19bc56ed7d82d2c0a586286300fd56c41367a/images/hub/Dockerfile#L66-L67) instead of COPY prevents duplicating the wheels into the build image. That might have a bigger impact than the rest of the changes.